server/src/resources/product/product.repository.ts

```diff
@@ -0,0 +1,47 @@
+import { IProduct, ProductEntity } from './product.interfaces';
+import { Product } from './product.models';
+
+class ProductRepository {
+  getProducts = async (): Promise<ProductEntity[] | null> => {
+    const products = await Product.find();
+
+    return products;
+  }
+
+  getProduct = async (id: string): Promise<ProductEntity | null> => {
+    const product = await Product.findById({ id });
```

Comment:
This request not working:
![image](https://github.com/BlazarQSO/NodeJS/assets/16895020/3bb93801-a2f0-42b3-a13c-a06838ef87d5)